src/pages/MerchantGiftCards/MerchantGiftCards.tsx

```diff
@@ -0,0 +1,145 @@
+  const [loading, setLoading] = useState(true);
+  const [giftCards, setGiftCards] = useState<any | null>();
+  const [seller, setSeller] = useState<any | null>();
+
+  const renderAmount = (props) => {
+    return Intl.NumberFormat('en-US', {
+      style: 'currency',
+      currency: 'USD',
+      maximumSignificantDigits: 4,
+    }).format(props.value / 100);
+  };
+
+  const renderDate = (props) => {
+    return new Date(props.value).toISOString().substring(0, 10);
+  };
+
+  const fields = [
+    {
+      name: 'seller_gift_card_id',
+      displayName: 'Voucher Code \n 礼品券号码',
+      inputFilterable: true,
+      sortable: true,
+    },
+    {
+      name: 'email',
+      displayName: 'Email \n 电子邮件',
+      inputFilterable: true,
+      sortable: true,
+    },
+    {
+      name: 'value',
+      displayName: 'Original Amount \n 购买金额',
+      inputFilterable: true,
+      sortable: true,
+      render: renderAmount,
+    },
+    {
+      name: 'created_at',
+      displayName: 'Date Purchased \n 购买日期',
+      inputFilterable: true,
+      sortable: true,
+      render: renderDate,
+    },
+    /* {
+      name: 'expiration',
+      displayName: 'Expiration',
+      inputFilterable: true,
+      sortable: true,
+    },
+    Uncomment if we want expiration */
+  ];
+
+  const fetchData = async () => {
+    try {
+      const giftCardData = (
+        await getMerchantGiftCards(metadata.sellerId, metadata.secretId)
+      ).data;
+      const sellerData = (await getSeller(metadata.sellerId)).data;
+
+      setGiftCards(giftCardData);
+      setSeller(sellerData);
+    } catch {
+      setError(true);
+    }
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    fetchData();
+
+    // eslint-disable-next-line
+  }, []);
+
+  return (
+    <>
+      {loading ? (
+        <Loader isPage={true} />
+      ) : error ? (
+        <ErrorPage menuOpen={false} />
+      ) : (
+        <>
+          <div className={styles.header}>
+            <h1>Voucher Tracker <span className={styles.noBreak}>礼品券记录</span></h1>
+            <h2>{seller.name}</h2>
+          </div>
+          <div className={styles.metadataHeader}>
+            <div className={styles.metadataBlock}>
+              <h1>LAST UPDATED <span className={styles.noBreak}>上次更新时间</span></h1>
+              <h2>{renderDate({ value: new Date() })}</h2>
+            </div>
+            <div className={styles.metadataBlock}>
+              <h1>ACTIVE VOUCHERS <span className={styles.noBreak}>可使用的礼品券数量</span></h1>
+              <h2>{giftCards && giftCards.length}</h2>
+            </div>
+            <div className={styles.metadataBlock}>
+              <h1>TOTAL BALANCE <span className={styles.noBreak}>总结余</span></h1>
+              <h2>
+                {renderAmount({
+                  value: giftCards.reduce((acc, cur) => acc + cur.value, 0),
+                })}
+              </h2>
+            </div>
+          </div>
+          <FilterableTable
```

Comment:
maybe we can make our own Filter Table so we can use one less dependency, ngl alot of these props are giving me anxiety LOL. But this is good for now too.

Reply:
😬  It's definitely a little anxiety-inducing. I think given that there's a discussion on how important gift cards might actually turn out to be, I thought I'd try to reduce the time to at least an MVP. I think when we want to make these fields editable (and add buttons, etc) we should either roll our own, or find a more modern table system.